src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -509,7 +509,8 @@ private static ImmutableArray<CompletionItemWithHighlight> GetHighlightedList(Li
             ImmutableArray<CompletionFilterWithState> filters)
         {
             // See which filters might be enabled based on the typed code
-            var textFilteredFilters = filteredList.SelectMany(n => n.VSCompletionItem.Filters).ToHashSet();
+            var _ = PooledHashSet<CompletionFilter>.GetInstance(out var textFilteredFilters);
```

Comment:
Does this usage allow the hash set to be returned to the pool?

Reply:
Yea, it returns the disposer (but i think this might not be available in compiler layer)

http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/J/s/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/ObjectPools/PooledHashSet.cs/PooledHashSet.cs,9